.github/compatibility-test-matrices/compat.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/python3
+import argparse
+import os
+import json
+
+# JSON keys to search in.
+KEYS = ("chain-a", "chain-b")
+# Toggle if required. Indent = 4 matches jq.
+DUMP_ARGS = {"indent": 4, "sort_keys": False, "ensure_ascii": False}
+# Suggestions for recent and new versions.
+SUGGESTION = "for example, v4.3.0 or 4.3.0"
+
+def find_json_files(directory, ignores=(".",)):
+    """ Find JSON files in a directory. By default, ignore hidden directories."""
+    for root, dirs, files in os.walk(directory):
+        dirs[:] = (d for d in dirs if not d.startswith(ignores))
+        for file_ in files:
+            if file_.endswith(".json"):
+                yield os.path.join(root, file_)
+
+def has_release_version(json_file, keys, version):
+    """ Check if the json file has the version in question. """
+    return any(version in row for row in (json_file[key] for key in keys))
+
+def update_version(json_file, keys, args):
```

Comment:
technically handled in the parse args, code, but maybe we can raise a `ValueError` if the operation is unknown in this function too just to be safe and explode when bad inputs are used.